src/main/java/com/powsybl/openloadflow/ac/AcloadFlowEngine.java

```diff
@@ -123,11 +122,15 @@ public AcLoadFlowResult run() {
                 .map(outerLoop -> Pair.of(outerLoop, new AcOuterLoopContext(context.getNetwork())))
                 .collect(Collectors.toList());
 
+        Optional<DistributedSlackContextData> distributedSlackContextDataOptional = Optional.empty();
         // outer loops initialization
         for (var outerLoopAndContext : outerLoopsAndContexts) {
             var outerLoop = outerLoopAndContext.getLeft();
             var outerLoopContext = outerLoopAndContext.getRight();
             outerLoop.initialize(outerLoopContext);
+            if (outerLoop instanceof DistributedSlackOuterLoop) {
```

Comment:
Maybe it would be simpler to do this after the NR run when we iterate over outer loops for calling cleanup method ?

Reply:
right, thanks